config/initializers/custom_initializers.rb

```diff
@@ -9,4 +9,10 @@
     f = File.join(ap['path'], 'config', 'custom_models.rb')
     eval(File.read(f)) if File.exist?(f)
   end
+
+  # This can be overridden in the app initializer to wrap the sleep and delete_file in an async job
+  CamaleonCmsUploader.delete_block do |settings, cama_uploader, file_key|
+    sleep(settings[:temporal_time])
```

Comment:
Wouldn't this block execution until the specified time passes? The old version of this code that I found wrapped this in a separate thread to avoid blocking. That approach woudn't work well for longer timeframes though, because the thread would not survive a server restart.

Since this feature hasn't been enabled for the last 8 years, I think the default should be a no-op, until/unless we add an ActiveJob dependency in the future.

If you think we should provide guidance on implementing this feature, we could replace the contents of this method with some commented-out code, for example:
```ruby
# This can be overridden in the app initializer to wrap the delete_file in an async job
CamaleonCmsUploader.delete_block do |temporal_time, file_key|
  # CamaleonDeleteFileJob.set(wait: temporal_time).perform_later(file_key)
  # # put this in app/jobs/camaleon_delete_file_job.rb:
  # include CamaleonCms::UploaderHelper
  # cama_uploader.delete_file(file_key)
end
```

That code is just an idea, I haven't tested it. If we go with that approach, we'd probably want to use the sleep approach in the tests as you have them now, which is fantatstic.

This also feels like we're pretty far off from the original purpose of the PR, so maybe we should pursue this separately if it needs more work.

Reply:
No more work needed - I just made the default code with sleep to work only in test environment, and added your comments with the async job.

So, it seems to be ready.

Reply:
Excellent, thanks for all the work on this!